pkg/protocol/http1/req/request.go

```diff
@@ -358,6 +358,15 @@ func ContinueReadBody(req *protocol.Request, r network.Reader, maxBodySize int,
 		req.Reset()
 		return err
 	}
+
+	if req.Header.ContentLength() == -1 {
+		err = ext.ReadTrailer(&req.Header.Trailer, r)
+		if err != nil && err != io.EOF {
+			return err
+		}
+		return nil
```

Comment:
If trailer exists, then req.Header.ContentLength won't be set. It is not the same with before